src/server/bg_services/lifecycle.js

```diff
@@ -23,9 +23,51 @@ function get_expiration_timestamp(expiration) {
     }
 }
 
+async function delete_expired_objects(system, bucket, rule, reply_objects) {
+    /* 
+    Note that expired delete-markers are also deleted by this query
+    since the lack of filter_delete_markers means the function's internal
```

Comment:
Hi Utkarsh, missing `filter_delete_markers` and `latest_versions` is deleting all the version including delete marker for the expired objects. We have a issue for that. https://issues.redhat.com/browse/DFBUGS-1468

Reply:
Adding the PR link here for a backlink - https://github.com/noobaa/noobaa-core/pull/8899